integration-tests/src/test/java/oracle/kubernetes/operator/ITServerDiscovery.java

```diff
@@ -0,0 +1,541 @@
+
+    logger.info("Check the newly created managed server is running");
+    int replicas = 3;
+    varifyPodReady(replicas);
+
+    // restore the test env
+    int msDecrNum = 1;
+    scaleDownAndVarify(msDecrNum);
+
+    logger.info("SUCCESS - " + testMethodName);
+  }
+
+  /**
+   * Verify that after bounce Operator, it reconnects to a pre-configed and newly started MS
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testOPReconnToNewMS() throws Exception {
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+
+    logger.info("Stop the Operator");
+    stopOperator();
+
+    String cmd = "kubectl apply -f " + testDomainYamlFile;
+    logger.info("Start a new managed server, managed-server3 using command:\n" + cmd);
+    ExecResult result = ExecCommand.exec(cmd);
+
+    logger.info("Start the Operator");
+    startOperator();
+
+    logger.info("Check the newly created managed-server3 is running");
+    int replicas = 3;
+    varifyPodReady(replicas);
+
+    // restore the test env
+    int msDecrNum = 1;
+    scaleDownAndVarify(msDecrNum);
+
+    logger.info("SUCCESS - " + testMethodName);
+  }
+
+  /**
+   * Verify that after bounce Operator, it discovers running servers and connect to them in place.
+   * It's able to scale down the cluster
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testOPReconnToRunningMSAndScaleDown() throws Exception {
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+
+    logger.info("Restart the Operator");
+    stopOperator();
+    startOperator();
+
+    int msDecrNum = 1;
+    scaleDownAndVarify(msDecrNum);
+
+    logger.info("Check managed-server1 is still running");
+    int replicas = 1;
+    varifyPodReady(replicas);
+
+    // restore the test env
+    int msIncrNum = 1;
+    scaleUpAndVarify(msDecrNum);
+
+    logger.info("SUCCESS - " + testMethodName);
+  }
+
+  /**
+   * Verify that after bounce Operator, it discovers running servers and connect to them in place.
+   * It's able to scale up the cluster
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testOPReconnToRunningMSAndScaleUp() throws Exception {
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+
+    logger.info("Restart the Operator");
+    stopOperator();
+    startOperator();
+
+    int msIncrNum = 1;
+    scaleUpAndVarify(msIncrNum);
+
+    // restore the test env
+    int msDecrNum = 1;
+    scaleDownAndVarify(msDecrNum);
+
+    logger.info("SUCCESS - " + testMethodName);
+  }
+
+  /**
+   * Verify that after bounce Operator, it discovers running servers and connect to them in place.
+   * It's able to scale up the cluster when domain has App deployed
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testOPReconnToRunningMSWApp() throws Exception {
```

Comment:
Server discovery test has nothing to do with WebApplication Deployment. We have test that covers the usecase for accessing WebApplication after scaling in ITOperatorTTest.
So the  the testmethod testOPReconnToRunningMSWApp look redundant and can be avoided

Reply:
will remove testOPReconnToRunningMSWApp